build.rs

```diff
@@ -501,10 +501,19 @@ fn windows_hardcoded_cross_compile(
 
     let python_version = PythonVersion { major, minor };
 
+    let is_extension_module = env::var_os("CARGO_FEATURE_EXTENSION_MODULE").is_some();
+    let no_enable_shared = env::var_os("PYO3_CROSS_NO_ENABLE_SHARED").is_some();
```

Comment:
This variable probably needs to be added here to ensure cross compilation gets enabled when it is set:
https://github.com/PyO3/pyo3/blob/7eb3aa3fd276d6f94684f4a3aff4c080b9816954/build.rs#L145-L147

Reply:
I can add it, but this variable is only used when cross-compiling for Windows, which requires `PYO3_CROSS_LIB_DIR` to be set as well.
The cross compilation mode is also automatically entered when the `--target` argument is different from the host triple, and `PYO3_CROSS` variable can be set to force it. As a user, I'd rather expect `PYO3_CROSS_LIB_DIR` and `PYO3_CROSS_PYTHON_VERSION` not do to anything when not cross-compiling.

Reply:
I think for now probably a good idea to add it, mostly just because cargo has issues with cross compiling and forcing it might make debugging easier in some cases.

Reply:
Added and rebased on `main`.